hw/drivers/crypto/src/mbedtls_aes_alt.c

```diff
@@ -53,15 +54,59 @@ mbedtls_aes_setkey_enc(mbedtls_aes_context *ctx, const unsigned char *key,
         return 0;
     }
 
-    return -1;
+    return MBEDTLS_ERR_AES_INVALID_KEY_LENGTH;
+}
+
+int
+mbedtls_aes_validate_input(mbedtls_aes_context *ctx, const unsigned char *key,
+                           unsigned int keybits)
+{
+    assert(ctx);
+
+    if (key == NULL) {
+        return MBEDTLS_ERR_AES_INVALID_KEY_LENGTH;
+    }
+
+    return 0;
+}
+
+int
+mbedtls_aes_setkey_enc(mbedtls_aes_context *ctx, const unsigned char *key,
+                       unsigned int keybits)
+{
+    int ret = mbedtls_aes_validate_input(ctx, key, keybits);
+
+    if (ret != 0) {
+        return ret;
+    }
+
+    return mbedtls_aes_setkey(ctx, key, keybits);
+}
+
+int
+mbedtls_aes_setkey_dec(mbedtls_aes_context *ctx, const unsigned char *key,
+                       unsigned int keybits)
+{
+    int ret = mbedtls_aes_validate_input(ctx, key, keybits);
+
+    if (ret != 0) {
+        return ret;
+    }
+
+    return mbedtls_aes_setkey(ctx, key, keybits);
 }
 
 int
 mbedtls_aes_crypt_ecb(mbedtls_aes_context *ctx, int mode,
-        const unsigned char input[16], unsigned char output[16])
+                      const unsigned char input[16], unsigned char output[16])
 {
-    return crypto_encrypt_aes_ecb(ctx->crypto, ctx->key, ctx->keylen,
-            (const uint8_t *)input, (uint8_t *)output, AES_BLOCK_LEN);
+    int ret = crypto_encrypt_aes_ecb(ctx->crypto, ctx->key, ctx->keylen,
+                                     (const uint8_t *)input, (uint8_t *)output, AES_BLOCK_LEN);
+    if (ret == AES_BLOCK_LEN) {
```

Comment:
I think `return (ret == AES_BLOCK_LEN) ? 0 : -1;` would look a bit better in this case.

Reply:
done